packages/jaeger-ui/src/components/Monitor/ServicesView/serviceGraph.test.js

```diff
@@ -39,61 +39,66 @@ describe('<ServiceGraph>', () => {
     expect(wrapper.length).toBe(1);
   });
 
-  it('Loading indicator is displayed', () => {
-    expect(wrapper).toMatchSnapshot();
+  it('displays loading indicator when loading', () => {
+    expect(wrapper.find('LoadingIndicator')).not.toBeNull();
   });
 
   it('Loading indicator is displayed when xDomain is empty', () => {
     wrapper.setProps({ ...props, xDomain: [], loading: false });
-    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find('LoadingIndicator')).not.toBeNull();
   });
 
-  it('"No data" displayed', () => {
+  it('displays "No Data" message when metricsData is null', () => {
     wrapper.setProps({ ...props, loading: false });
-    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.text()).toContain('No Data');
   });
 
-  it('"Couldn’t fetch data" displayed', () => {
+  it('displays "Couldn’t fetch data" message when error is present', () => {
     wrapper.setProps({ ...props, loading: false, error: new Error('API Error') });
-    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.text()).toContain('Couldn’t fetch data');
   });
 
-  it('Base graph should be displayed ', () => {
-    wrapper.setProps({ ...props, loading: false, metricsData: serviceMetrics.service_call_rate });
-    expect(wrapper).toMatchSnapshot();
+  it('displays base graph when metricsData is provided', () => {
+    wrapper.setProps({
+      ...props,
+      loading: false,
+      metricsData: serviceMetrics.service_call_rate,
+    });
+    expect(wrapper.find('ComposedChart').exists()).toBe(true);
   });
 
-  it('Base graph with legends should be displayed', () => {
+  it('displays base graph with legends when showLegend is true', () => {
     wrapper.setProps({
       ...props,
       loading: false,
       metricsData: serviceMetrics.service_call_rate,
       showLegend: true,
     });
-    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find('Legend').exists()).toBe(true);
   });
 
-  it('Base graph with horizontal lines should be displayed', () => {
+  it('displays base graph with horizontal lines when showHorizontalLines is true', () => {
     wrapper.setProps({
       ...props,
       loading: false,
       metricsData: serviceMetrics.service_call_rate,
       showHorizontalLines: true,
     });
-    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find('CartesianGrid[horizontal=true]').exists()).toBe(true);
   });
 
-  it('Base graph with custom color should be displayed', () => {
+  it('displays base graph with custom color', () => {
     wrapper.setProps({
       ...props,
       loading: false,
       metricsData: serviceMetrics.service_call_rate,
       color: 'AAAAAA',
     });
-    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find('Area[stroke="AAAAAA"]').exists()).toBe(true);
+    expect(wrapper.find('Line[stroke="AAAAAA"]').exists()).toBe(true);
   });
 
-  it('Crosshair map test', () => {
+  it('Tooltip test', () => {
```

Comment:
This changes what the test was looking for. Is the crosshair pointer not available? Or we can't test for it? I assume it's just part of the style.